internal/api/dto/subscription.go

```diff
@@ -16,7 +16,12 @@ import (
 )
 
 type CreateSubscriptionRequest struct {
-	CustomerID         string               `json:"customer_id" validate:"required"`
+
+	//CustomerID is the flexprice customer id
+	//CustomerID is prioritized over ExternalCustomerID(if present)
+	CustomerID string `json:"customer_id"`
+	//ExternalCustomerID is the external customer id provided by the developer
+	ExternalCustomerID string               `json:"external_customer_id"`
```

Comment:
```suggestion
	// customer_id is the flexprice customer id
	// and it is prioritized over external_customer_id in case both are provided
	CustomerID string `json:"customer_id"`
	//external_customer_id is the customer id in your DB and must be same as what you provided as external_id while creating the customer in flexprice.
	ExternalCustomerID string               `json:"external_customer_id"`
```